Source/UI/Validator/PhoneNumberValidator.swift

```diff
@@ -29,11 +35,32 @@ class PhoneNumberValidator: Validator {
 
     func formatForDisplay(text: String) -> String {
         do {
-            let formattedNumber = try phoneKit.parse(text)
+            let formattedNumber = try phoneKit.parse(text,
+                                                     withRegion: countryCode)
             return phoneKit.format(formattedNumber, toType: .international)
         } catch {
             return text
         }
     }
 
 }
+
+// MARK: Singleton behaviour
+extension PhoneNumberValidator {
+
+    static var shared: PhoneNumberValidator {
+        if let instance {
+            return instance
+        }
+        let newValidator = PhoneNumberValidator()
+        instance = newValidator
+        return newValidator
+    }
+    static private var instance: PhoneNumberValidator?
```

Comment:
I accept that computed properties can be messy and look less like a variable declaration and more like a function, but in this case they are strictly organised by access level and role:
- internal property
- private property
- internal method

I'd rather be consistent with formatting, so other alternative may be to make `shared` a function, but then we'll invoke it as `PhoneNumberValidator.shared().___` everywhere, which will trigger flashbacks of consuming Obj-C singletons in Swift.

We may also break apart the logic from the dynamic getter into a private function, making it look cleaner:
```Swift

    static var shared: PhoneNumberValidator {
        Self.getShared()
    }
    static private var instance: PhoneNumberValidator?

    /// Will deallocate the singleton source
    static func removeSingleton() {
        instance = nil
    }

    private static getShared() -> PhoneNumberValidator {
        if let instance {
            return instance
        }
        let newValidator = PhoneNumberValidator()
        instance = newValidator
        return newValidator
    }
```
I would however argue that this last version needlessly splits logic into "another function" and just adds noise.

So personally I'd suggest keeping it as is, since it continues to structure files consistently with project, but happy to discuss further if you feel another path may be beneficial.